.gitlab-ci.yml

```diff
@@ -0,0 +1,168 @@
+    - apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+    # Prepare shell environment
+    - source venv/bin/activate
+    - export PATH=$PATH:portal/node_modules/.bin
+    # Login to AWS container registry
+    - aws ecr get-login-password --region eu-central-1 | docker login --username AWS --password-stdin 322480324822.dkr.ecr.eu-central-1.amazonaws.com
+
+install-job:
+  stage: install
+  before_script:
+    - apt-get update
+    - apt-get install -y nodejs npm python3.10-venv libxml2-dev libxslt-dev python3.10-dev
+  script:
+    - python3 --version
+    - python3 -m venv venv --copies --upgrade-deps
+    - source venv/bin/activate
+    - pip install setuptools==58
+    - pip install -r requirements.txt
+    - invoke aws.sync-repository-state --no-profile
+    - invoke prepare-builds --commit $CI_COMMIT_SHA
+    - cd portal
+    - npm ci
+
+lint-job:
+  stage: lint-and-test
+  extends: .tooling-job
+  script:
+    - flake8 .
+    - cd portal
+    - npm run lint
+
+service-test-job:
+  stage: lint-and-test
+  services:
+    - name: postgres:12.2-alpine
+    - name: docker.elastic.co/elasticsearch/elasticsearch:7.9.3
+      alias: elasticsearch
+      command: [ "bin/elasticsearch", "-Expack.security.enabled=false", "-Ediscovery.type=single-node" ]
+  extends: .tooling-job
+  script:
+    - APPLICATION_PROJECT=edusources invoke test.service
+    - APPLICATION_PROJECT=nppo invoke test.service
+
+harvester-test-job:
+  stage: lint-and-test
+  services:
+    - name: postgres:12.2-alpine
+  extends: .tooling-job
+  script:
+    - APPLICATION_PROJECT=edusources invoke test.harvester
+    - APPLICATION_PROJECT=nppo invoke test.harvester
+
+build-job:
+  stage: build
+  extends: .docker-job
+  script:
+    - invoke aws.build harvester --commit $CI_COMMIT_SHA
+    - invoke aws.build service --commit $CI_COMMIT_SHA
+    - invoke aws.push harvester --commit $CI_COMMIT_SHA
+    - invoke aws.push service --commit $CI_COMMIT_SHA
+
+integration-test-job:
+  stage: integration-test
+  services:
+    - name: $AWS_CONTAINER_REGISTRY/harvester:$CI_COMMIT_SHA
+      alias: harvester
+    - name: $AWS_CONTAINER_REGISTRY/search-portal:$CI_COMMIT_SHA
+      alias: service
+    - name: docker:dind
+    - name: postgres:12.2-alpine
+    - name: docker.elastic.co/elasticsearch/elasticsearch:7.9.3
+      alias: elasticsearch
+      command: [ "bin/elasticsearch", "-Expack.security.enabled=false", "-Ediscovery.type=single-node" ]
+  extends: .docker-job
+  variables:
+    POL_ELASTIC_SEARCH_ENABLE_DECOMPOUND_ANALYZERS: 0
+  script:
+    - docker ps
+    - cypress install
+    - invoke srv.setup-postgres --host postgres
+    - invoke hrv.setup-postgres --host postgres
+    - invoke hrv.index-dataset-version localhost -d test
+    - curl http://harvester:8888/
+    - curl http://service:8000/health
+    # TODO: we need Cypress to use service:8080 instead of localhost:8000
```

Comment:
do we have a ticket to do this, so we wont forget?

Reply:
https://www.pivotaltracker.com/story/show/182261850